primitives/src/menubar.rs

```diff
@@ -0,0 +1,223 @@
+        recent_focus: Signal::new(0),
+        current_focus: Signal::new(None),
+    });
+
+    rsx! {
+        div {
+            role: "menubar",
+            "data-disabled": (props.disabled)(),
+
+            onfocusout: move |_| ctx.set_focus(None),
+            ..props.attributes,
+
+            {props.children}
+        }
+    }
+}
+
+#[derive(Props, Clone, PartialEq)]
+pub struct MenubarMenuProps {
+    index: usize,
+
+    #[props(default)]
+    disabled: ReadOnlySignal<bool>,
+
+    #[props(extends = GlobalAttributes)]
+    attributes: Vec<Attribute>,
+    children: Element,
+}
+
+#[component]
+pub fn MenubarMenu(props: MenubarMenuProps) -> Element {
+    let mut ctx: MenubarContext = use_context();
+
+    use_effect(move || {
+        ctx.menu_count += 1;
+    });
+
+    let is_open = use_memo(move || (ctx.open_menu)() == Some(props.index));
+    let tab_index = use_memo(move || {
+        if (ctx.current_focus)() == Some(props.index) {
+            "0"
+        } else {
+            "-1"
+        }
+    });
+
+    rsx! {
+        div {
+            role: "menu",
+            tabindex: tab_index,
+            "data-state": if is_open() { "open" } else { "closed" },
+            "data-disabled": (ctx.disabled)() || (props.disabled)(),
+
+            onclick: move |_| {
+                if !(ctx.disabled)() && !(props.disabled)() {
+                    let new_open = if is_open() { None } else { Some(props.index) };
+                    ctx.set_open_menu.call(new_open);
+                }
+            },
+
+            onfocus: move |_| ctx.set_focus(Some(props.index)),
+
+            onkeydown: move |event: Event<KeyboardData>| {
+                let mut prevent_default = true;
+                match event.key() {
+                    Key::Enter => {
+                        if !(ctx.disabled)() && !(props.disabled)() {
+                            let new_open = if is_open() { None } else { Some(props.index) };
+                            ctx.set_open_menu.call(new_open);
+                        }
+                    }
+                    Key::Escape => ctx.set_open_menu.call(None),
+                    Key::ArrowLeft => ctx.focus_prev(),
+                    Key::ArrowRight => ctx.focus_next(),
+                    Key::Home => ctx.focus_first(),
+                    Key::End => ctx.focus_last(),
+                    _ => prevent_default = false,
+                }
+                if prevent_default {
+                    event.prevent_default();
+                }
+            },
+
+            ..props.attributes,
+            {props.children}
+        }
+    }
+}
+
+#[derive(Props, Clone, PartialEq)]
+pub struct MenubarTriggerProps {
+    #[props(extends = GlobalAttributes)]
+    attributes: Vec<Attribute>,
+    children: Element,
+}
+
+#[component]
+pub fn MenubarTrigger(props: MenubarTriggerProps) -> Element {
+    rsx! {
+        button { r#type: "button", ..props.attributes, {props.children} }
```

Comment:
No action nescessary here, but I thought I'd share that `type` (without r#) is valid now too!